proposals/NNNN-noncopyable-generics.md

```diff
@@ -731,234 +490,119 @@ let _: any Copyable = t.peelOneTopping() // signature <Self: Copyable>
 let u: any Pizza & ~Copyable = ... // signature <Self: Pizza>
 let _: any ~Copyable = u.peelOneTopping() // signature <Self>
 ```
-For associated types within a protocol erased to an existential preserve conformances to a marker protocol like `Copyable`. So when calling `peelOneTopping` on an `any Pizza`, an `any ~Copyable` value is returned instead of `any Copyable`, which is equivalent to `Any`.
+For associated types within a protocol erased to an existential preserve conformances to a marker protocol like `Copyable`.
+So when calling `peelOneTopping` on an `any Pizza`, an `any ~Copyable` value is returned instead of `any Copyable`, which is equivalent to `Any`.
 
 ### Scoping rule
 
-An inverse constraint like `~Copyable` can only be applied to a type parameter within the same scope as the constraint:
+A constraint suppression like `~Copyable` can only be applied to a type parameter within the same scope as the constraint:
 
 ```swift
 struct S<T> { // signature: <T: Copyable>
   func f() where T: ~Copyable // signature: <T>
-  // error: cannot add inverse constraint 'T: ~Copyable' on generic parameter 'T' defined in outer scope
+  // error: cannot suppress constraint 'T: ~Copyable' on generic parameter 'T' defined in outer scope
 }
 ```
 
-The rationale is that an outer generic context, like `S<T>`, already requires that `T` is `Copyable`. Removing that `Copyable` requirement for the nested generic context `S<T>.f` is useless, as there will never be a noncopyable value substituted for `S<T>`. The same logic applies to mututally-scoped contexts:
+The rationale is that an outer generic context, like `S<T>`, already requires that `T` is `Copyable`.
+Removing that `Copyable` requirement for the nested generic context `S<T>.f` is useless, as there will never be a noncopyable value substituted for `S<T>`.
+The same logic applies to mututally-scoped contexts:
 
 ```swift
 protocol P {
-  // error: cannot add inverse constraint 'Self.Alice: ~Copyable' on generic parameter 'Self.Alice' defined in outer scope
+  // error: cannot suppress constraint 'Self.Alice: ~Copyable' on generic parameter 'Self.Alice' defined in outer scope
   associatedtype Bob where Alice: ~Copyable
   associatedtype Alice where Bob: ~Copyable
-  // error: cannot add inverse constraint 'Self.Bob: ~Copyable' on generic parameter 'Self.Bob' defined in outer scope
+  // error: cannot suppress constraint 'Self.Bob: ~Copyable' on generic parameter 'Self.Bob' defined in outer scope
 }
 ```
 
-
-<!-- ### `AnyObject`
-
-TODO: can classes be cast to `any ~Copyable`? If so, then it seems fine to
-permit an `AnyObject` to be cast to `any ~Copyable`. 
-
-TODO: is this legal? `func f<T>(_ t: T) where T: AnyObject, T: ~Copyable {}` -->
-
-<!-- ## Effect on ABI stability -->
-
-<!-- ## Effect on API resilience -->
```

Comment:
These are worth uncommenting and keeping around, since we need to answer / fill-in these sections.